docs/section23/Rule23-16.md

```diff
@@ -5,42 +5,43 @@
 **Appendix G Section:** Section 23 Air-side  
 **90.1 Section Reference:** Section G3.1.3.19 Preheat Coils (Systems 5 through 8)  
 **Data Lookup:** None  
 **Evaluation Context:** Building  
 
 **Applicability Checks:**  
 
-1. B-RMI is modeled with at least one air-side system that is Type-5, 6, 7, 8, 7a, 8a, 5b, 6b, 7b, 8b, 7c.  
+1. B-RMR is modeled with at least one air-side system that is Type-5, 6, 7, 8, 7a, 8a, 5b, 6b, 7b, 8b, 7c.  
 
 **Function Calls:**  
 
 1. get_baseline_system_types()
-2. get_component_by_id()
+2. baseline_system_type_compare()
+3. get_component_by_id()
 
 **Applicability Checks:**  
+- create a list of the target system types: `target_system_types = [HVAC_SYS.SYS_5,HVAC_SYS.SYS_6,HVAC_SYS.SYS_7,HVAC_SYS.SYS_8]`
+- Get B-RMR system types: `baseline_hvac_system_dict = get_baseline_system_types(B-RMR)`
 
-- Get B-RMI system types: `baseline_hvac_system_dict = get_baseline_system_types(B-RMI)`
+  - Check if B-RMR is modeled with at least one air-side system that is Type-5, 6, 7, 8, 7a, 8a, 5b, 6b, 7b, 8b, 7c, continue to rule logic: `if any(baseline_system_type_compare(system_type, target_sys_type, false) for system_type in baseline_hvac_system_dict.keys() for target_system_type in target_system_types): CHECK RULE LOGIC`
 
-  - Check if B-RMI is modeled with at least one air-side system that is Type-5, 6, 7, 8, 7a, 8a, 5b, 6b, 7b, 8b, 7c, continue to rule logic: `if any(sys_type in baseline_hvac_system_dict.keys() for sys_type in ["SYS-5", "SYS-6", "SYS-7", "SYS-8", "SYS-7A", "SYS-8A", "SYS-5B", "SYS-6B", "SYS-7B", "SYS-8B", "SYS-7C"]): CHECK_RULE_LOGIC`
-
-  - Else, rule is not applicable to B-RMI: `else: RULE_NOT_APPLICABLE`
+  - Else, rule is not applicable to B-RMR: `else: RULE_NOT_APPLICABLE`
 
 ## Rule Logic:  
-- Create a list of the ids of the hvac systems that are one of the targeted system types: `eligible_systems = []`
-- For each baseline system type in the baseline_hvac_system_dict: `for baseline_system_type in baseline_hvac_system_dict:`
-  - check if the baseline_system_type is one of 6, 8, 8a, 6b, 8b: `if baseline_system_type in ["SYS-5", "SYS-6", "SYS-7", "SYS-8", "SYS-7A", "SYS-8A", "SYS-5B", "SYS-6B", "SYS-7B", "SYS-8B", "SYS-7C"]:`
-    - add the hvac_ids for this system type to the list of eligible_systems: `eligible_systems += baseline_hvac_system_dict[baseline_system_type]`
 
-- For each zone in B_RMI: `for zone_b in B_RMI...zones:`
+- make a list of eligible hvac systems: `eligible_systems = []`
+- loop through the system types in baseline_hvac_system_dict: `for system_type in baseline_hvac_system_dict:`
+  - check if it is one of the target baseline system types: `if any(baseline_system_type_compare(system_type,target_system_type,false) for target_system_type in target_system_types):`
+    - add the hvac system ids to the list of eligible systems: `eligible_systems = eligible_systems + baseline_hvac_system_dict[system_type]`
+- For each zone in B_RMR: `for zone_b in B_RMR...zones:`
```

Comment:
Why not use the existing function get_dict_of_zones_and_terminal_units_served_by_hvac_sys instead to get zones that are served by each eligible system and then follow logic in lines 46 and 48 for these zones?

Reply:
updated